classes/itemlist.php

```diff
@@ -791,41 +791,58 @@ public function display_relations_table() {
     }
 
     /**
-     * Adds elements to an array starting from initial conditions.
-     *
-     * Conditions are array('hidden' => 0) OR array('reserved' => 0)
-     * Add to $nodelist all the children that will inherit the parent condition
+     * Get the recursive list of children of a specific item.
+     * This list counts children and children of children for as much generation as it is.
      *
-     * @param integer $nodelist
-     * @param array $additionalcondition
-     * @return void
+     * @param int $oldestparent
+     * @param array $where
+     * @return object $childrenitems
      */
-    public function add_child_node(&$nodelist, $additionalcondition) {
+    public function item_get_children($oldestparent=null, $where=null) {
         global $DB;
 
-        if (!is_array($additionalcondition)) {
-            $a = 'add_child_node';
+        if (empty($oldestparent)) {
+            $oldestparent = $this->itemid;
```

Comment:
I'm not sure I understand this. Why the itemlist class (that handles lists/collections of items), has a `$this->itemid` attribute? What is the content of that itemid? If it's the id of the root/base item in the structure... why not also call it that way? The same than in the previous comment. Just for consistence, that helps to understand that both are the same thing (the main item in a tree structure of items).